script/testing/util/test_server_v2.py

```diff
@@ -0,0 +1,223 @@
+
+    def __init__(self, args):
+        """ Locations and misc. variable initialization """
+        # clean up the command line args
+        self.args = {k: v for k, v in args.items() if v}
+
+        # server output
+        self.db_output_file = self.args.get("db_output_file",
+                                            constants.DEFAULT_DB_OUTPUT_FILE)
+
+        # set the DB path
+        self.set_db_path()
+        self.db_process = None
+
+        # db server location
+        self.db_host = self.args.get("db_host", constants.DEFAULT_DB_HOST)
+        self.db_port = self.args.get("db_port", constants.DEFAULT_DB_PORT)
+
+        return
+
+    def run_pre_suite(self):
+        pass
+
+    def run_post_suite(self):
+        pass
+
+    def set_db_path(self):
+        """ location of db server, relative to this script """
+
+        # builds on Jenkins are in build/<build_type>
+        # but CLion creates cmake-build-<build_type>/<build_type>
+        # determine what we have and set the server path accordingly
+        bin_name = constants.DEFAULT_DB_BIN
+        build_type = self.args.get("build_type", "")
+        path_list = [
+            os.path.join(constants.DIR_REPO, "build", build_type),
+            os.path.join(constants.DIR_REPO,
+                         "cmake-build-{}".format(build_type), build_type)
+        ]
+        for dir in path_list:
+            path = os.path.join(dir, bin_name)
+            if os.path.exists(path):
+                self.db_path = path
+                return
+
+        msg = "No DB binary found in {}".format(path_list)
+        raise RuntimeError(msg)
+
+    def check_db_binary(self):
+        """ Check that a Db binary is available """
+        if not os.path.exists(self.db_path):
+            abs_path = os.path.abspath(self.db_path)
+            msg = "No DB binary found at {}".format(abs_path)
+            raise RuntimeError(msg)
+        return
+
+    def run_db(self):
+        """ Start the DB server """
+
+        # Allow ourselves to try to restart the DBMS multiple times
+        for attempt in range(constants.DB_START_ATTEMPTS):
+            # Kill any other terrier processes that our listening on our target port
+            for other_pid in check_port(self.db_port):
+                print(
+                    "Killing existing server instance listening on port {} [PID={}]"
+                    .format(self.db_port, other_pid))
+                os.kill(other_pid, signal.SIGKILL)
+            # FOR
+
+            self.db_output_fd = open(self.db_output_file, "w+")
+            self.db_process = subprocess.Popen(self.db_path,
+                                               stdout=self.db_output_fd,
+                                               stderr=self.db_output_fd)
+            try:
+                self.wait_for_db()
+                break
+            except:
+                self.stop_db()
+                print("+" * 100)
+                print("DATABASE OUTPUT")
```

Comment:
just popped into my mind that we will be using `logging` for output information in the future right? Ben has been working on this in another task.

Reply:
add TODO